### DISCUSSION TITLE
What is the recommended way to find out which URL a request came from?

### DISCUSSION THREAD

**Discussion by silvenon** [remix-run/remix#5421] [Q&A] [answered]
I'm making a request to another route using `useFetcher`, so from that route's action I would like to return a different response based on the URL the request came from. I'm providing a non-JS experience from a different route, in which case I'd like it to redirect.

`request.referrer` doesn't exist by default, is there a way to provide it via `useFetcher`? However,`request.headers.get('referrer')` seems to work, so is it ok?

**Comment by clgeoio** [comment, marked as answer]
You could add a header on the request `x-source: foobar` and then have the action perform something based on the header.

**Reply by silvenon** [reply]
That sounds like the way to go, yeah. I'll see what others say…
